zeppelin-interpreter/src/main/java/org/apache/zeppelin/scheduler/Job.java

```diff
@@ -182,22 +182,16 @@ public void run() {
       this.exception = null;
       errorMessage = null;
       dateFinished = new Date();
-      progressUpdator.terminate();
-    } catch (NullPointerException e) {
-      LOGGER.error("Job failed", e);
-      progressUpdator.terminate();
-      this.exception = e;
-      setResult(e.getMessage());
-      errorMessage = getStack(e);
-      dateFinished = new Date();
     } catch (Throwable e) {
       LOGGER.error("Job failed", e);
-      progressUpdator.terminate();
       this.exception = e;
       setResult(e.getMessage());
       errorMessage = getStack(e);
       dateFinished = new Date();
     } finally {
+      if (progressUpdator != null) {
+        progressUpdator.interrupt();
```

Comment:
`terminate` was called in both cases: on line 185 in case of success and on line 195 in catch block. So it is better to place it in `finally` to handle both cases.